core/store/src/trie/iterator.rs

```diff
@@ -367,15 +375,24 @@ impl<'a> Iterator for TrieIterator<'a> {
     fn next(&mut self) -> Option<Self::Item> {
         loop {
             let iter_step = self.iter_step()?;
+
+            let can_process = match self.max_depth {
+                Some(max_depth) => self.key_nibbles.len() <= max_depth,
+                None => true,
+            };
+            if !can_process {
+                continue;
```

Comment:
I don't understand the code perfectly to check if this works as expected, but

if you do `continue` here, `self.trail.pop();` at line 390 will not happen. Wouldn't it violate some invariants?